pkg/services/ngalert/api/api.go

```diff
@@ -86,7 +87,7 @@ func (api *API) RegisterAPIEndpoints(m *metrics.API) {
 	api.RegisterAlertmanagerApiEndpoints(NewForkedAM(
 		api.DatasourceCache,
 		NewLotexAM(proxy, logger),
-		&AlertmanagerSrv{store: api.AlertingStore, mam: api.MultiOrgAlertmanager, secrets: api.SecretsService, log: logger, ac: api.AccessControl},
+		&AlertmanagerSrv{secrets: api.SecretsService, log: logger, ac: api.AccessControl, configs: services.NewAlertmanagerConfigService(api.MultiOrgAlertmanager, api.SecretsService, api.AlertingStore, logger)},
```

Comment:
The new AlertmanagerConfigService now encapsulates the MultiOrgAlertmanager and AlertingStore. The component dependency graph is a bit deeper, but less complex in any single location.